apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java

```diff
@@ -559,6 +575,51 @@ public ClassLoader getApplicationClassLoader() {
         }
     }
 
+    public byte[] serialize() {
+        byte[] result = new byte[SERIALIZED_LENGTH];
+        serialize(result);
+        return result;
+    }
+
+    public void serialize(byte[] buffer) {
```

Comment:
Maybe switch to using the binary header functionalities for serialization (`fillOutgoingTraceParentBinaryHeader`) and deserialization (`asChildOf(byte[])`), with an extended byte array where you separately serialize/deserialize the clock and service name? Just so we don't need to maintain both